### ISSUE TITLE
Option to not parse the content of specific containers?

### ISSUE THREAD

**Issue by calaoa** [Nevenall/remark-containers#4]
Hi @Nevenall, all, Would it make sense to add an option for not parsing as Markdown specific containers content (hence leaving the content as is), unless there is such an option already? A typical use case would be the following: a container contains some YAML data to be interpreted later on in the display process, e.g. by a JavaScript application executed by the browser: in that case, it could be more convenient for the user to not surround that content with escaping characters (\`\`\`), just typing as follows:

```
::: hero
img: banner.jpg
title: "My title"
buttons: [{id: "id1", label: "label1"}, {id: "id2", label: "label2"}]
:::
```
What do you think?
Cheers and thumbs up

**Comment by Nevenall** [comment]
Sounds like a helpful feature! 
Currently you can use the options object with `default: false` but you'd have to explicitly define all the containers you do want parsed.

**Comment by calaoa** [comment]
Thanks a lot Nevenall, I gave a try to `default: false` with the code below, but the node seems to have been turned into a Markdown one already (I would expect a text node rather than a list, no?), not sure what I'm doing wrong?

## Code

```
unified()
  .use(remarkParse)
  .use(remarkAttr)
  .use(containers, {
    default: false,
    custom: [{
      type: 'hero',
      element: 'hero',
      transform: function (node: any, config: any, tokenize: any) {
        console.log(node);
      }
    }]
  })
```

## Input

```
::: hero
- charlie
- _aloha_
:::
```
## Output

```
{
  type: 'hero',
  data: { hName: 'hero' },
  children: [
    {
      type: 'list',
      ordered: false,
      start: null,
      spread: false,
      children: [Array],
      position: [Position]
    }
  ]
}

```

**Comment by Nevenall** [comment]
When default is false only the specified custom containers will be parsed so, just remove the hero custom container. If you have other containers you DO want parsed you'll have to specify each of them as a custom container.

**Comment by Nevenall** [comment]
I'll see what I can do about adding an option to skip parsing specified containers.

**Comment by calaoa** [comment]
Nevenall, Thank you for your help, I misunderstood the role of that parameter. I gave a try to setting it to `false` indeed.  The unified + parse + containers + remark2rehype + stringify chain applied to the sample above produces the following then:

```
<p>::: hero</p>
<ul>
<li>charlie</li>
<li><em>aloha</em><br>
:::</li>
</ul>
```

Which can be useful indeed, however what I'm trying to obtain would be this instead:

```
<hero>
- charlie
- _aloha_
</hero>
```

Would there be a way to achieve this with the current implementation or does is require further coding? What do you think?

**Comment by Nevenall** [comment]
Oh, I see now!
The contents of the container always get parsed currently, but you could undo it with a custom transform.

```
{
  type: 'hero',
  element: 'hero',
  transform: function (node, config, tokenize) {
    // node will be the abstract tree for the html
    // you can translate the <ul> and <li> back into markdown. 
    // I bet there is a remark function to do that.

  }
}
```

I can add an option to disable parsing contents. 

What if it was a keyword you can use for any container, like this:

```
::: noparse hero
- charlie
- _aloha_
:::
```

**Comment by calaoa** [comment]
Indeed, reverse parsing could make it, I had actually considered this path until I figured it would be costly for large containers... Preventing the parsing completely sounds better, a `noparse` keyword would be nice, however what do you think about something possibly more generic? Let's imagine a container that is meant to display a diff between two text blocks, accepting various layout or algorithmic options:  such options could be passed as a parameter to the container transformer, in addition to the one meant to prevent the Markdown parsing, e.g. something along this line as a JSON object at the end of the declaration:

```
::: diff {"md": "false", "layout":"side-by-side"}
- _aloha_
- charlie
---
- aloha
- charlie
::: 
```

I looked into what others are suggesting or implementing for such wiki macros:

## CommonMark directives / plugins proposal

```
::: name [inline-content] {key=val}
contents, which are sometimes further block elements
:::
``` 
Reference: https://talk.commonmark.org/t/generic-directives-plugins-syntax/444

## MakeCode macros

```
### ~avatar [class]

[content]

### ~
```
Reference: https://makecode.com/writing-docs/macros



## XWiki

```
{{chart source="inline" type="<type>" params="<params>" title="<title>" height="<height>" width="<width>"}}
// Tabular data used as an input for the chart
{{/chart}}
```
Reference: [XWiki Chart Macro](https://extensions.xwiki.org/xwiki/bin/view/Extension/Chart%20Macro)

**Comment by Nevenall** [comment]
Ok! Added the `noparse` flag and published version 1.2.0

now you can use 

``` 
::: noparse hero
- charlie
- _aloha_
:::
```
to get 
```
<hero>
- charlie
- _aloha_
</hero>
```

**Comment by calaoa** [comment]
Awesome! Thumbs up
